plugins/modules/ndo_fabric_span_session.py

```diff
@@ -0,0 +1,705 @@
+        type: dict
+        suboptions:
+          name:
+            description:
+            - The name of the destination port channel.
+            type: str
+            required: true
+          template:
+            description:
+            - The name of the template that contains the destination port channel.
+            - This parameter or O(destination_port_channel.port_channel.template_id) is required.
+            type: str
+          template_id:
+            description:
+            - The ID of the  that contains the destination port channel.
+            - This parameter or O(destination_port_channel.port_channel.template) is required.
+            type: str
+  state:
+    description:
+    - Use C(absent) for removing.
+    - Use C(query) for listing an object or multiple objects.
+    - Use C(present) for creating or updating.
+    type: str
+    choices: [ absent, query, present ]
+    default: query
+notes:
+- The O(template) must exist before using this module in your playbook.
+  Use M(cisco.mso.ndo_template) to create the Fabric Monitoring Access Policy template.
+- The O(destination_epg.epg) must exist before using it with this module in your playbook.
+  Use M(cisco.mso.mso_schema_template_anp_epg) to create the EPG.
+- The O(destination_port_channel.port_channel) must exist before using it with this module in your playbook.
+  Use M(cisco.mso.ndo_port_channel_interface) to create the Fabric resource port channel interface.
+seealso:
+- module: cisco.mso.ndo_template
+- module: cisco.mso.mso_schema_template_anp_epg
+- module: cisco.mso.ndo_port_channel_interface
+extends_documentation_fragment: cisco.mso.modules
+"""
+
+EXAMPLES = r"""
+- name: Create EPG - Fabric SPAN Session
+  cisco.mso.ndo_fabric_span_session:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_test
+    name: ansible_test_epg
+    destination_epg:
+      epg:
+        schema: ansible_test
+        template: Template1
+        anp: Anp1
+        name: EPG1
+      destination_ip: "1.1.1.1"
+      source_ip_prefix: "2.2.2.2"
+    state: present
+  register: create_epg_span_session
+
+- name: Create Port - Fabric SPAN Session
+  cisco.mso.ndo_fabric_span_session:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_test
+    name: ansible_test_port
+    destination_port:
+      port:
+        node: 101
+        path: "eth1/1"
+    state: present
+
+- name: Create Port Channel - Fabric SPAN Session
+  cisco.mso.ndo_fabric_span_session:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_test
+    name: ansible_test_pc
+    destination_port_channel:
+      port_channel:
+        template: ansible_test
+        name: ansible_test_resource_pc_1
+    state: present
+
+- name: Update/Change EPG - Fabric SPAN Session to Port Fabric SPAN Session
```

Comment:
why use update/change and update below, is update not enough what we do in other modules also?

Reply:
Done!